lua/luasnip/extras/treesitter_postfix.lua

```diff
@@ -0,0 +1,320 @@
+			local bufnr = vim.api.nvim_win_get_buf(0)
+			local cursor = vim.api.nvim_win_get_cursor(0)
+			local parser, source =
+				ts.reparse_buffer_after_removing_match(bufnr, matched_trigger)
+			if parser == nil or source == nil then
+				return nil
+			end
+			local ret = real_resolver(
+				snippet,
+				line_to_cursor,
+				matched_trigger,
+				captures,
+				parser,
+				source,
+				bufnr,
+				{ cursor[1] - 1, cursor[2] - #matched_trigger - 1 }
+			)
+			parser:destroy()
+			return ret
+		end
+	end
+
+	return function(snippet, line_to_cursor, matched_trigger, captures)
+		local bufnr = vim.api.nvim_win_get_buf(0)
+		local cursor = vim.api.nvim_win_get_cursor(0)
+		local parser, source = vim.treesitter.get_parser(bufnr), bufnr
+		if parser == nil or source == nil then
+			return nil
+		end
+		return real_resolver(
+			snippet,
+			line_to_cursor,
+			matched_trigger,
+			captures,
+			parser,
+			source,
+			bufnr,
+			{ cursor[1] - 1, cursor[2] - #matched_trigger - 1 }
+		)
+	end
+end
+
+local function generate_resolve_expand_param(match_tsnode, user_resolver)
+	---@param snippet any
+	---@param line_to_cursor string
+	---@param matched_trigger string
+	---@param captures any
+	---@param parser LanguageTree
+	---@param source number|string
+	---@param bufnr number
+	---@param pos { [1]: number, [2]: number }
+	return function(
+		snippet,
+		line_to_cursor,
+		matched_trigger,
+		captures,
+		parser,
+		source,
+		bufnr,
+		pos
+	)
+		local cursor = vim.api.nvim_win_get_cursor(0)
+		local ts_parser = ts.TSParser.new(bufnr, parser, source)
+		if ts_parser == nil then
+			return
+		end
+
+		local row, col = unpack(pos)
+
+		local match_node
+		if type(match_tsnode) ~= "function" then
+			match_node = generate_match_tsnode_func(match_tsnode)
+		else
+			match_node = match_tsnode
+		end
+		if match_node == nil then
+			return nil
+		end
+
+		local match_result = match_node(ts_parser, { row, col })
+		if match_result == nil then
+			return nil
+		end
+
+		local matches = {}
+
+		for _, info in ipairs(match_result.matches) do
+			local start_row, start_col, text =
+				ts_parser:get_node_text(info.node, cursor, matched_trigger)
+			if matches[info.capture_name] == nil then
+				matches[info.capture_name] = {}
+			end
+			table.insert(matches[info.capture_name], {
+				start = { start_row, start_col },
+				text = text,
+				type = info.node:type(),
+			})
+		end
+
+		local start_row, start_col, best_match_text = ts_parser:get_node_text(
+			match_result.best_match,
+			cursor,
+			matched_trigger
+		)
+
+		local ret = {
+			trigger = matched_trigger,
+			captures = captures,
+			clear_region = {
+				from = {
+					start_row,
+					start_col,
+				},
+				to = {
+					cursor[1] - 1,
+					cursor[2],
+				},
+			},
+			env_override = {
+				TREESITTER_MATCHES = matches,
+				TREESITTER_BEST_MATCH = {
+					start = { start_col, start_col },
+					text = best_match_text,
+					type = match_result.best_match:type(),
+				},
```

Comment:
Mhmm passing this data is cool, but `env` is meant to only contain strings or a list of strings..  (`$TM_SOME_VAR` in lsp-snippets directly accesses `env.TM_SOME_VAR`).
I'm not generally opposed to passing other things as well, but we'll have to 
* provide at least the text as strings as well (so, separately, s.t. it can be used by lsp-snippets)
* clearly communicate what is in `env`
* make sure the current code handling `env` doesn't assume there's only strings in there.